src/Util/Network.php

```diff
@@ -559,9 +559,8 @@ public static function isUrlAllowed($url)
 	 */
 	public static function isUrlBlocked($url)
 	{
-		$h = @parse_url($url);
-
-		if (! $h) {
+		$host = @parse_url($url, PHP_URL_HOST);
```

Comment:
Good catch, `parse_url()` without 2nd parameter does return the full array.

OT: Maybe we should remove `@`? I find it not good muting PHP functions. Instead maybe switch to a proper class (composer again) which allows validation?

Reply:
Adding the `@` operator to `parse_url()` calls seems to have been the norm in this codebase, we probably should check if it's too verbose without now. We deal with a lot of unreachable/malformed URLs, I wouldn't be surprised if it throws Warning messages all the time.

Reply:
My idea here is, that we give parsing URLs away to a suitable library (class) that is preferable a 3rd-party one (`composer require foo/bar`). The library  should then have a validation method like `isValid()` or so which can be used prior parsing it. If still parsing is done on an invalid URL, the parsing method should throw e.g. a `MalformedUrlException`.

Yes, I'm a Java developer, too. :-)